sw/device/silicon_creator/lib/drivers/otbn.h

```diff
@@ -0,0 +1,155 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_OTBN_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_OTBN_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct otbn {
+  mmio_region_t base_addr;
+} otbn_t;
+
+/**
+ * Start the execution of the application loaded into OTBN at the start address.
+ *
+ * @param otbn OTBN instance
+ * @param start_addr The IMEM byte address to start the execution at.
+ * @return `kErrorOtbInvalidArgument` if `start_addr` is invalid, `kErrorOk`
+ *         otherwise.
+ */
+rom_error_t otbn_start(const otbn_t *otbn, unsigned int start_addr);
+
+/**
+ * Is OTBN busy executing an application?
+ *
+ * @param otbn OTBN instance
+ * @param[out] busy OTBN is busy
+ */
+void otbn_is_busy(const otbn_t *otbn, bool *busy);
+
+/**
+ * OTBN Errors
+ *
+ * OTBN uses a bitfield to indicate which errors have been seen. Multiple errors
+ * can be seen at the same time. This enum gives the individual bits that may be
+ * set for different errors.
+ */
+typedef enum otbn_err_bits {
+  kOtbnErrBitsNoError = 0,
+  /** Load or store to invalid address. */
+  kOtbnErrBitsBadDataAddr = (1 << 0),
+  /** Instruction fetch from invalid address. */
+  kOtbnErrBitsBadInsnAddr = (1 << 1),
+  /** Call stack underflow or overflow. */
+  kOtbnErrBitsCallStack = (1 << 2),
+  /** Illegal instruction execution attempted */
+  kOtbnErrBitsIllegalInsn = (1 << 3),
+  /** LOOP[I] related error */
+  kOtbnErrBitsLoop = (1 << 4),
+  /** Error seen in Imem read */
+  kOtbnErrBitsFatalImem = (1 << 5),
+  /** Error seen in Dmem read */
+  kOtbnErrBitsFatalDmem = (1 << 6),
+  /** Error seen in RF read */
+  kOtbnErrBitsFatalReg = (1 << 7)
+} otbn_err_bits_t;
```

Comment:
I think we should add static assertions for these to make sure that they match `OTBN_ERR_BITS_*_BIT`